src/lib.rs

```diff
@@ -1471,10 +1538,252 @@ decl_module! {
 
             Ok(())
         }
+
+        /// Implement on_initialize for data migration
+        fn on_initialize(n: T::BlockNumber) {
```

Comment:
There are two points to this approach that perhaps could be changed

**1. Doing step by step removal of data from the old module:**
This is much slower than doing one singular deletion of all the storage of a module at the end of the migration. This is possible using 

https://substrate.dev/rustdocs/master/sp_io/storage/fn.clear_prefix.html

It may even be desirable to avoid doing any such deletion automatically and do the final cleanup in a separate upgrade, or at a later time.

**2. Mutating the old module to keep track of how far the migration has gone:**

This disrupts the read state of the module during migration. Remember, people may still be _reading_ the forum and using it for consumption during the migration. This will be broken if there is continuous mutation. Its not such a big problem in a very fast migration, but the longer it takes, the bigger the problem becomes. It also is a little less clear to understand how the progress is encoded, compared to having explicit state variables for the migration that are separate, and in the standalone migration module.

Reply:
1. I will remove the code to remove old during the migration.
clear_prefix is a new method to me, I try give the input as b"NewForum" and b"NewForum CategoryById", but seems it doesn't work in my test. Could you please tell me how to give the correct input?
2. I assume the old forum should be completely disable after the migration fork. I will keep the old data during migration and put the code related to migration into a new module.

Reply:
1.  I believe it mainly depends on the module name of the old forum module, as this is a prefix of key of all storage nodes associated with this module. I do not know the exact string though, perhaps asking in the Substrate matrix chat is the best alternative.

Reply:
I will ask parity people how to do it.

Reply:
it is the right approach.
system::kill_prefix(
      twox_128("Staking") ++ twox_128("SlashingSpans")
)